tests/core/component.test.js

```diff
@@ -1036,13 +990,39 @@ suite('Component', function () {
       el.setAttribute('dummy', direction);
       sinon.assert.calledTwice(updateStub);
       // oldData passed to the update method.
-      assert.equal(updateStub.getCalls()[0].args[0].x, undefined);
-      assert.equal(updateStub.getCalls()[0].args[0].y, undefined);
-      assert.equal(updateStub.getCalls()[0].args[0].z, undefined);
-      assert.deepEqual(updateStub.getCalls()[1].args[0], {x: 1, y: 1, z: 1});
```

Comment:
This test has been re-arranged since the captured values are by reference. Properties on oldData are only valid for the duration of `update`. Now the relevant assertions are taken during the `update`.

On master this tests 'works' because the referenced object hasn't been changed yet. Making the chain longer (say three updates), it also shows the same problem. Now this is a small behavioural change, but users shouldn't be keeping references to either data or oldData even before this change.